staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/validator.go

```diff
@@ -193,3 +200,36 @@ func (a customResourceValidator) ValidateScaleStatus(ctx context.Context, obj *u
 
 	return allErrs
 }
+
+func (a customResourceValidator) validateStatusObservedGeneration(ctx context.Context, obj, old *unstructured.Unstructured) field.ErrorList {
+	var allErrs field.ErrorList
+
+	observedGeneration, exists, err := unstructured.NestedInt64(obj.UnstructuredContent(), "status", "observedGeneration")
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("status.observedGeneration"), observedGeneration, err.Error()))
```

Comment:
how does this deal with existing persisted data, since this is tightening validation?